pkg/encoding/v1/encoding.go

```diff
@@ -481,5 +555,46 @@ func (d *Decoder) Decode(data []byte) (*object.OpenCompose, error) {
 		openCompose.Volumes = append(openCompose.Volumes, ov)
 	}
 
+	// convert secrets
+	for _, secret := range v1.Secrets {
+		var os object.Secret
+		os.Name = string(secret.Name)
+		for _, secData := range secret.Data {
+
+			// If the secret has been provided through a file
+			// and
+			// if it is a relative path
+			// and
+			// the OpenCompose input file is either STDIN or file path
+			// then
+			// convert it to an absolute file path.
+			// This will fail if input file is URL
+			if (secData.File != nil) && !filepath.IsAbs(*secData.File) {
```

Comment:
I have explained it in the comment above. Basically, the relative paths of the secret is getting converted to the absolute path so it can be read from a file.